### DISCUSSION TITLE
Internet Access

### DISCUSSION THREAD

**Discussion by gcormier9** [roleoroleo/sonoff-hack#47] [Q&A]
Hi,

I am trying to setup port forwarding on on my Internet router in order to access cameras from Internet. To make a short story, I have kept the default ports on 2 cameras and I have setup port 81 to be forwarded to camera #1, port 1000 for ONVIF and port 554 to be forwarded to camera #1, port 554 for RTSP. Using this configuration, I can sucesfully access camera #1 from Internet. However, I can't access camera #2 since I haven't find a way to change the RTSP port. Can you provide some guidance? Is there a better way to access multiple cameras from Internet?

Thanks!

**Comment by roleoroleo** [comment]
The port is defined inside the sonoff fw.
But you could change the external port when you create your forwarding rule:

Camera 1
- http: external 81 -> cam1 81
- onvif: external 1000 -> cam1 1000
- rtsp: externam 554 -> cam1 554

Camera 2
- http: external 82 -> cam2 81
- onvif: external 1001 -> cam2 1000
- rtsp: externam 555 -> cam2 554

**Reply by gcormier9** [reply]
Do you know if the rtsp port defined inside sonoff fw can be changed?

In regards to port forwarding, I don't believe your method will work since for camera #2, ONVIF will redirect the RTSP stream on port 554 since it is not aware of the port 555.

**Reply by roleoroleo** [reply]
> Do you know if the rtsp port defined inside sonoff fw can be changed?

I think it's not possible.

> In regards to port forwarding, I don't believe your method will work since for camera #2, ONVIF will redirect the RTSP stream on port 554 since it is not aware of the port 555.

You are right, if you use onvif it's not possible.
You have to use directly rtsp.

**Comment by roleoroleo** [comment]
Probably I found a solution to change the port.
I will try it.

**Reply by gcormier9** [reply]
Amazing, please let me know if you find something.

**Reply by roleoroleo** [reply]
See this commit:
https://github.com/roleoroleo/sonoff-hack/commit/622b6732425eca0859f415aea2a345abf105a622

**Reply by gcormier9** [reply]
Wow! That was fast! Thank you very much! Do you plan to create a new release soon?

**Reply by roleoroleo** [reply]
No, but I could create a zip file if you want to try it.

**Comment by gcormier9** [comment]
I'm interested to test it!

**Reply by roleoroleo** [reply]
Decompress it and overwrite your old files:
[sonoff-hack.tar.gz](https://github.com/roleoroleo/sonoff-hack/files/6520804/sonoff-hack.tar.gz)

**Comment by gcormier9** [comment]
Thanks! I'll give it a try!

**Reply by gcormier9** [reply]
I've just tested it. Based on commit https://github.com/roleoroleo/sonoff-hack/commit/622b6732425eca0859f415aea2a345abf105a622, is it possible [sonoff-hack.tar.gz](https://github.com/roleoroleo/sonoff-hack/files/6520804/sonoff-hack.tar.gz) is missing some files:
 - /sonoff-hack/etc/system.conf
 - /sonoff-hack/script/check_conf.sh
 - /sonoff-hack/script/system.sh

After adding those files, I am able to change the RTSP port. However, I have noticed I cannot change the port on the web interface. It only work if I make the change on `/sonoff-hack/etc/system.conf`. I don't know much about ONVIF protocol, but I've also noticed that if I connect to ONVIF port it still try to use RTSP port 554. However, my ONVIF application allow me to override the RTSP port, so it works fine. I just don't know if this is an expected behavior or not...

Anyway, thanks for your help/support on this! It's much appreciated!

**Comment by roleoroleo** [comment]
If I'm not wrong the last archive contains system.sh and check_conf.sh.
system.conf is not needed.
https://github.com/roleoroleo/sonoff-hack/files/6520804/sonoff-hack.tar.gz
With this patch you should be able to change the port from the web interface.

About ONVIF, it's only a protocol descriptor, it doesn't implement streaming protocols but it use RTSP.
